### ISSUE TITLE
Build errors with containerized instructions 

### ISSUE THREAD

**Issue by funnelfiasco** [SeaGL/seagl.github.io#274]
I'm having some trouble building the site. I'm not Docker or Ruby fluent enough to figure it out on my own, but it seems like something relatively straightforward for someone who knows what they're doing. :-)

~~~~
[1074 bcotton@normal ~/devel/seagl/seagl.github.io ]$ podman build --tag 'seagldev' '.'
STEP 1: FROM ruby:2.6
STEP 2: RUN apt-get update && apt-get -y dist-upgrade
--> Using cache c86285afdf1a73b94cc9c329f4e9b7b0cc9609e368d393673585fe6b40e78c86
--> c86285afdf1
STEP 3: RUN bundle config --global frozen 1
--> Using cache d09ac802a15904a0feb9448e395eb0543177bd07b3563891fc13ccabeb88fc58
--> d09ac802a15
STEP 4: WORKDIR /usr/src/app
--> Using cache ecf2e8e9ea6bfd1f52f5d1f51a18b30b15728603a6cb37537586552888184e10
--> ecf2e8e9ea6
STEP 5: COPY Gemfile Gemfile.lock ./
--> Using cache f0981ba4ba82c2a113472d938b1b0d3d3aaead6dc452630f54ab2063bad541b2
--> f0981ba4ba8
STEP 6: RUN bundle install
/usr/local/lib/ruby/2.6.0/rubygems.rb:283:in `find_spec_for_exe': Could not find 'bundler' (2.1.4) required by your /usr/src/app/Gemfile.lock. (Gem::GemNotFoundException)
To update to the latest version installed on your system, run `bundle update --bundler`.
To install the missing version, run `gem install bundler:2.1.4`
        from /usr/local/lib/ruby/2.6.0/rubygems.rb:302:in `activate_bin_path'
        from /usr/local/bin/bundle:23:in `<main>'
Error: error building at STEP "RUN bundle install": error while running runtime: exit status 1
~~~~

If I add a `bundler update --bundler` step, I get:

~~~~
STEP 6: RUN bundle update --bundler
You must use Bundler 2 or greater with this lockfile.
Error: error building at STEP "RUN bundle update --bundler": error while running runtime: exit status 20
~~~~

I can build with the bare metal by replacing the system ruby 3.0 with the ruby 2.7 module. (I'm not sure if that will affect anything else on my system)

**Comment by AndrewKvalheim** [comment]
Not sure about the decision to bail so strictly on Bundler version mismatches, but the first thing I see is GitHub Pages [uses](https://pages.github.com/versions/) Ruby 2.7 and the Dockerfile hasn’t been updated ([oops](https://github.com/SeaGL/seagl.github.io/commit/c38fdd846ba7092bc38d49a9640dd2aef9439562)) to use that yet. [Doing that](https://github.com/SeaGL/seagl.github.io/commit/29506b94e950184f3e467d15ccae8ebbb7ad8a24) incidentally dodges this issue for now.

**Comment by funnelfiasco** [comment]
With your fix, the build runs, but the serving bombs

~~~~
[1104 bcotton@normal ~/devel/seagl/seagl.github.io ]$ podman run --rm --interactive --tty --publish '4000:4000' --volume "$PWD:/usr/src/app" 'seagldev'
Configuration file: none
            Source: /usr/src/app
       Destination: /usr/src/app/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
jekyll 3.9.0 | Error:  Permission denied @ dir_initialize - .
Traceback (most recent call last):
        37: from /usr/local/bin/bundle:23:in `<main>'
        36: from /usr/local/bin/bundle:23:in `load'
        35: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/libexec/bundle:34:in `<top (required)>'
        34: from /usr/local/lib/ruby/2.7.0/bundler/friendly_errors.rb:123:in `with_friendly_errors'
        33: from /usr/local/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/libexec/bundle:46:in `block in <top (required)>'
        32: from /usr/local/lib/ruby/2.7.0/bundler/cli.rb:24:in `start'
        31: from /usr/local/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/base.rb:476:in `start'
        30: from /usr/local/lib/ruby/2.7.0/bundler/cli.rb:30:in `dispatch'
        29: from /usr/local/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor.rb:399:in `dispatch'
        28: from /usr/local/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        27: from /usr/local/lib/ruby/2.7.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        26: from /usr/local/lib/ruby/2.7.0/bundler/cli.rb:476:in `exec'
        25: from /usr/local/lib/ruby/2.7.0/bundler/cli/exec.rb:28:in `run'
        24: from /usr/local/lib/ruby/2.7.0/bundler/cli/exec.rb:63:in `kernel_load'
        23: from /usr/local/lib/ruby/2.7.0/bundler/cli/exec.rb:63:in `load'
        22: from /usr/local/bundle/bin/jekyll:23:in `<top (required)>'
        21: from /usr/local/bundle/bin/jekyll:23:in `load'
        20: from /usr/local/bundle/gems/jekyll-3.9.0/exe/jekyll:15:in `<top (required)>'
        19: from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
        18: from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
        17: from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
        16: from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
        15: from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
        14: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/commands/serve.rb:75:in `block (2 levels) in init_with_program'
        13: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/commands/serve.rb:93:in `start'
        12: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/commands/serve.rb:93:in `each'
        11: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/commands/serve.rb:93:in `block in start'
        10: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/commands/build.rb:36:in `process'
         9: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/commands/build.rb:65:in `build'
         8: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/command.rb:28:in `process_site'
         7: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/site.rb:69:in `process'
         6: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/site.rb:164:in `read'
         5: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/reader.rb:16:in `read'
         4: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/reader.rb:42:in `read_directories'
         3: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/reader.rb:42:in `chdir'
         2: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/reader.rb:42:in `block in read_directories'
         1: from /usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/reader.rb:42:in `entries'
/usr/local/bundle/gems/jekyll-3.9.0/lib/jekyll/reader.rb:42:in `open': Permission denied @ dir_initialize - . (Errno::EACCES)
~~~~

**Comment by AndrewKvalheim** [comment]
Hmm, works for me. Maybe bad permissions in `_site` from a past run with Docker? With our current instructions everything it generates is owned by root.

**Comment by funnelfiasco** [comment]
I removed `_site` and it doesn't re-create it. (Same error.) So I tried changing the `WORKDIR` in my Dockerfile to `/tmp`. That worked. I'm guessing because Podman is rootless it couldn't create `/usr/src/app`? But creating that directory doesn't change anything.

With `WORKDIR /tmp` everything works so :shrug:

**Comment by AndrewKvalheim** [comment]
FWIW this also works for me as-is using Podman on NixOS.

**Comment by funnelfiasco** [comment]
I tried it on a different machine and got a different failure at server-time:
`Could not locate Gemfile or .bundle/ directory`

But again, changing `WORKDIR` to `/tmp` made it work. So I'll just close this since it's apparently a _me_ problem. :-)

**Comment by AndrewKvalheim** [comment]
It looks like this is probably [a SELinux thing](https://github.com/containers/podman/blob/v3.4.2/troubleshooting.md#2-cant-use-volume-mount-get-permission-denied). I just tried in a Fedora Cloud Base VM and processes in the container have no read access to the mounted working directory until I set `:z`.

Not sure what the right way forward is but it would be nice to accommodate this environment.
